src/frameworks/foundation/ns_string.rs

```diff
@@ -435,6 +435,21 @@ pub const CLASSES: ClassExports = objc_classes! {
     new_string
 }
 
+- (id)lastPathComponent { // NSString*
+    let self_as_rust_string = &to_rust_string(env, this);
+    if self_as_rust_string == "/" {
+        let new_string = from_rust_string(env, "/".to_string());
+        autorelease(env, new_string);
+        new_string
+    } else {
+        let path = GuestPath::new(self_as_rust_string);
+        let file_name = path.file_name().unwrap_or("");
+        let new_string = from_rust_string(env, String::from(file_name));
+        autorelease(env, new_string);
+        new_string
+    }
+}
```

Comment:
Ah, sorry, the more I look at this, the more I notice it's not correct, and I'm not entirely sure how to fix it… are you okay if I just don't merge it for now?

Reply:
That's fine. In what ways is it incorrect?

Reply:
Just staring at https://developer.apple.com/documentation/foundation/nsstring/1416528-lastpathcomponent?language=objc, I think it's always incorrect to return `""` unless the original string is `""`? I also realised the implementation in `fs.rs` is probably wrong in edge cases like `/foo/` too.

It probably works okay most of the time but eventually someone (probably me) needs to write unit tests for all the path component edge cases in `fs.rs` and make sure they all match Apple's behaviour.

Reply:
Yeah, you're absolutely right. I wrote some quick Swift code in a playground to test this with the example cases in the Rust docs[1], and NSString does things very differently. So this is something that will need to be revisited later.

```swift
import Foundation

let string: NSString = "foo.txt/.//"
// note: explicitly initialize as NSString instead of Swift String type

print("last path component: \(string.lastPathComponent)")

// This returns ".", while the corresponding Rust implementation will return "foo.txt".
```

[1]https://doc.rust-lang.org/std/path/struct.Path.html#method.file_name

Reply:
Oh no, I didn't even consider the problem of a mismatch between Rust's `Path` and `NSString`/`NSURL`! Thanks for pointing that out.

Reply:
A little update, in case it's interesting. The app I'm currently interested in also needed this method, so I've now done my own implementation, which I think is correct: als09c749c5f514a1cc73bb2ca296e14a4653d7f5a5